chichiwang/exercise-js
# Serve `/static` directory contents

## Description
Linked to Issue: #59
Move `/images` into a `/static` directory in the project root to be served statically. On build the `/static` directory will be copied into the `dist/` directory and served statically by the Express application.

Documentation has been updated with new npm scripts (`build:dist`, `build:static`) and information about how static files are handled within the project.

## Changes
* Added scripts to `package.json`
  * `build:dist`: Create a `/dist` directory in the project root
  * `prebuild:static`: Run `build:dist` to ensure `dist/` directory exists
  * `build:static`: Copy `/static` directory into the `/dist` directory
* Updated scripts in `package.json`
  * `prebuild:dev`: Run `clean` followed by `build:static`
  * `prebuild:prod`: Run `clean` followed by `build:static`
* Move `images/` directory into `static/images` directory
  * Move `images/favicon/site.webmanifest` file to `static/site.webmanifest`
* Update `server/index.ts` to statically serve a `/static` route from the `/static` directory
* Update `docs/README.md`
  * Add instruction for using the correct Node.js version in section `Starting a Project`
  * Add documentation for scripts `build:dist`, `build:static` to section `NPM Scripts`
  * Add section `Static Files`, add to table of contents in section `Developing Locally`

## Steps to QA
* Pull down and switch to this branch
* Run `npm run clean` and ensure `dist/` directory is gone
* Run `npm run build:static` to ensure `dist/` directory exists and contains a copy of the `static/` directory
* Run `npm run dev`/`npm run watch`/`npm run prod` and ensure the static directory is served statically (navigate browser to `/static/images/logo.webp` or any other file path and ensure it is served)
